clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
I see. This makes sense. One concern with making `UnknownServerException` a retriable exception is that client may retry unnecessarily. Maybe we can reduce unnecessary retries by distinguishing between `UnknownServerException` and `UnknownException`. An exception is an `UnknownException` if the error code is not found defined, .e.g. 0.11.0 client receives the error code for `KafkaStorageException`. Thus the client wouldn't have to retry if server sends the error code of the existing server unknown exception. And the client will only retry unnecessarily if its client's version is (temporarily) lower than server's version.

Does this sound OK? I can submit a separate patch for 0.11.0  and 0.10.2 branch.